sklearn/datasets/tests/test_openml.py

```diff
@@ -335,39 +156,182 @@ def _mock_urlopen(request, *args, **kwargs):
         context.setattr(sklearn.datasets._openml, "urlopen", _mock_urlopen)
 
 
+###############################################################################
+# Test the behaviour of `fetch_openml` depending of the input parameters.
+
+# Known failure of PyPy for OpenML. See the following issue:
+# https://github.com/scikit-learn/scikit-learn/issues/18906
+@fails_if_pypy
 @pytest.mark.parametrize(
-    "feature, expected_dtype",
+    "data_id, dataset_params, n_samples, n_features, n_targets",
     [
-        ({"data_type": "string", "number_of_missing_values": "0"}, object),
-        ({"data_type": "string", "number_of_missing_values": "1"}, object),
-        ({"data_type": "numeric", "number_of_missing_values": "0"}, np.float64),
-        ({"data_type": "numeric", "number_of_missing_values": "1"}, np.float64),
-        ({"data_type": "real", "number_of_missing_values": "0"}, np.float64),
-        ({"data_type": "real", "number_of_missing_values": "1"}, np.float64),
-        ({"data_type": "integer", "number_of_missing_values": "0"}, np.int64),
-        ({"data_type": "integer", "number_of_missing_values": "1"}, np.float64),
-        ({"data_type": "nominal", "number_of_missing_values": "0"}, "category"),
-        ({"data_type": "nominal", "number_of_missing_values": "1"}, "category"),
+        # iris
+        (61, {"data_id": 61}, 150, 4, 1),
+        (61, {"name": "iris", "version": 1}, 150, 4, 1),
+        # anneal
+        (2, {"data_id": 2}, 11, 38, 1),
+        (2, {"name": "anneal", "version": 1}, 11, 38, 1),
+        # cpu
+        (561, {"data_id": 561}, 209, 7, 1),
+        (561, {"name": "cpu", "version": 1}, 209, 7, 1),
+        # emotions
+        (40589, {"data_id": 40589}, 13, 72, 6),
+        # adult-census
+        (1119, {"data_id": 1119}, 10, 14, 1),
+        (1119, {"name": "adult-census"}, 10, 14, 1),
+        # miceprotein
+        (40966, {"data_id": 40966}, 7, 77, 1),
+        (40966, {"name": "MiceProtein"}, 7, 77, 1),
+        # titanic
+        (40945, {"data_id": 40945}, 1309, 13, 1),
     ],
 )
-def test_feature_to_dtype(feature, expected_dtype):
-    assert _feature_to_dtype(feature) == expected_dtype
+@pytest.mark.parametrize("parser", ["liac-arff", "pandas"])
```

Comment:
Instead of parametrizing by `parser`, can you please run both parsers on each dataset and check that the numeric values match, up to integer to float casting when necessary?